src/network/networkd-network.c

```diff
@@ -136,7 +140,17 @@ static int network_load_one(Manager *manager, const char *filename) {
         network->proxy_arp = -1;
         network->ipv6_accept_ra_use_dns = true;
 
-        r = config_parse(NULL, filename, file,
+        dropin_dirname = strjoin("/", network->name, ".network.d", NULL);
```

Comment:
you could just use strjoina() here (i.e. use stack allocation), after all the name is pretty much bounded. You can then drop _cleanup_free_ from the variable declaration, but keep the "const" for it...